<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="generator" content="Docusaurus v2.0.0-alpha.56">
<link rel="search" type="application/opensearchdescription+xml" title="PopGen.jl" href="/PopGen.jl/opensearch.xml"><title data-react-helmet="true">Hardy-Weinberg Equilibrium | PopGen.jl</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:title" content="Hardy-Weinberg Equilibrium | PopGen.jl"><meta data-react-helmet="true" name="description" content="Testing for Hardy-Weinberg Equilibrium (often abbreviated HWE) is a fairly common practice in population genetics. In a two-allele system, the HWE equation is defined as: $p^2 + 2pq + q^2 = 1$ , where $p$ is the frequency of the first allele and $q$ is the frequency of the second allele. The formula describes the frequency of all possible genotypes where"><meta data-react-helmet="true" property="og:description" content="Testing for Hardy-Weinberg Equilibrium (often abbreviated HWE) is a fairly common practice in population genetics. In a two-allele system, the HWE equation is defined as: $p^2 + 2pq + q^2 = 1$ , where $p$ is the frequency of the first allele and $q$ is the frequency of the second allele. The formula describes the frequency of all possible genotypes where"><meta data-react-helmet="true" property="og:url" content="https://pdimens.github.io/PopGen.jl/docs/analyses/hardyweinberg"><link data-react-helmet="true" rel="shortcut icon" href="/PopGen.jl/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://pdimens.github.io/PopGen.jl/docs/analyses/hardyweinberg"><link rel="stylesheet" href="/PopGen.jl/styles.ee3d76ba.css">
<link rel="preload" href="/PopGen.jl/styles.da9164d3.js" as="script">
<link rel="preload" href="/PopGen.jl/runtime~main.512265a5.js" as="script">
<link rel="preload" href="/PopGen.jl/main.5219d249.js" as="script">
<link rel="preload" href="/PopGen.jl/1.4e959812.js" as="script">
<link rel="preload" href="/PopGen.jl/2.23b37f7c.js" as="script">
<link rel="preload" href="/PopGen.jl/3.d4ef4d3f.js" as="script">
<link rel="preload" href="/PopGen.jl/1be78505.c8cbe7a4.js" as="script">
<link rel="preload" href="/PopGen.jl/67.d7d524a0.js" as="script">
<link rel="preload" href="/PopGen.jl/72e583ac.c15c2edd.js" as="script">
<link rel="preload" href="/PopGen.jl/731bcebf.924639df.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=window.matchMedia("(prefers-color-scheme: dark)"),n=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();null!==n?t(n):e.matches&&t("dark")}()</script><div id="__docusaurus">
<nav class="navbar navbar--light navbar--fixed-top navbarHideable_OaSq"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/PopGen.jl/"><img class="navbar__logo" src="/PopGen.jl/img/logo_icon.png" alt="PopGen.jl logo"><strong class="navbar__title">PopGen.jl</strong></a></div><div class="navbar__items navbar__items--right"><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/PopGen.jl/docs/">Docs</a><a class="navbar__item navbar__link" href="/PopGen.jl/docs/getting_started/about">About</a><a class="navbar__item navbar__link" href="/PopGen.jl/blog">Blog</a><a class="navbar__item navbar__link" href="/PopGen.jl/docs/getting_started/community">Get Involved</a><a href="https://github.com/pdimens/popgen.jl" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository">GitHub</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_1gtM"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_keGJ moon_1gwN"></span></div><div class="react-toggle-track-x"><span class="toggle_keGJ sun_3CPA"></span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div><div class="navbar__search"><div class="searchWrapper_-F3-"><span aria-label="expand searchbar" role="button" class="searchIconButton_1Dnz" tabindex="0"></span><input type="search" id="search_input_react" placeholder="Search" aria-label="Search" class="navbar__search-input searchInput_2eto"></div></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/PopGen.jl/"><img class="navbar__logo" src="/PopGen.jl/img/logo_icon.png" alt="PopGen.jl logo"><strong class="navbar__title">PopGen.jl</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/PopGen.jl/docs/">Docs</a></li><li class="menu__list-item"><a class="menu__link" href="/PopGen.jl/docs/getting_started/about">About</a></li><li class="menu__list-item"><a class="menu__link" href="/PopGen.jl/blog">Blog</a></li><li class="menu__list-item"><a class="menu__link" href="/PopGen.jl/docs/getting_started/community">Get Involved</a></li><li class="menu__list-item"><a href="https://github.com/pdimens/popgen.jl" target="_blank" rel="noopener noreferrer" class="menu__link header-github-link" aria-label="GitHub repository">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_1kjD"><div class="docSidebarContainer_1cYp"><div class="sidebar_1kLs sidebarWithHideableNavbar_1-U5"><a tabindex="-1" class="sidebarLogo_3ono" href="/PopGen.jl/"><img src="/PopGen.jl/img/logo_icon.png" alt="PopGen.jl logo"><strong>PopGen.jl</strong></a><div class="menu menu--responsive menu_w2sC"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_2vk4" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Getting Started</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/install">Installation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/julia_primer">A quick Julia primer</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/comparison">Comparison</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/types">PopObj and PopData types</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/other_types">Other data types</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/getting_started/datasets">Provided datasets</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">File I/O</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/io/file_import">Reading in data</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/io/file_write">Writing data</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/io/delimited">Delimited format</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/io/genepop">Genepop format</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/io/vcf">Variant Call Format</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Tutorial</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/tutorials/manipulate">Start here</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/tutorials/accessing">Accessing elements</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/tutorials/exclusion">Data Exclusion</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/tutorials/populations">Location and population data</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/tutorials/viewsort">Viewing and sorting</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Analyses</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/PopGen.jl/docs/analyses/hardyweinberg">Hardy-Weinberg Equilibrium</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/PopGen.jl/docs/analyses/relatedness">Relatedness (Kinship)</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">API</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/api">API</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/allelefreq">AlleleFreq.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/datasets">Datasets.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/delimited">Delimited.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/genepop">Genepop.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/hardyweinberg">HardyWeinberg.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/heterozygosity">Heterozygosity.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/ioutils">ioUtils.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/manipulate">Manipulate.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/read">Read.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/summaryinfo">SummaryInfo.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/types">Types.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/utils">Utils.jl</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/PopGen.jl/docs/api/variantcall">VariantCall.jl</a></li></ul></li></ul></div></div></div><main class="docMainContainer_FFX1"><div class="container padding-vert--lg docItemWrapper_1cc7"><div class="row"><div class="col docItemCol_2GOA"><div class="docItemContainer_2cwg"><article><header><h1 class="docTitle_1vWb">Hardy-Weinberg Equilibrium</h1></header><div class="markdown"><link rel="stylesheet" href="/PopGen.jl/katex/katex.min.css"><p>Testing for Hardy-Weinberg Equilibrium (often abbreviated <em>HWE</em>) is a fairly common practice in population genetics. In a two-allele system, the HWE equation is defined as: <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>p</mi><mn>2</mn></msup><mo>+</mo><mn>2</mn><mi>p</mi><mi>q</mi><mo>+</mo><msup><mi>q</mi><mn>2</mn></msup><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">p^2 + 2pq + q^2 = 1</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">p</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2222222222222222em"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em"></span></span><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em"></span><span class="mord">2</span><span class="mord mathdefault">p</span><span class="mord mathdefault" style="margin-right:0.03588em">q</span><span class="mspace" style="margin-right:0.2222222222222222em"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em"></span></span><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.03588em">q</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em"></span><span class="mord">1</span></span></span></span></span> , where <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi></mrow><annotation encoding="application/x-tex">p</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em"></span><span class="mord mathdefault">p</span></span></span></span></span> is the frequency of the first allele and <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>q</mi></mrow><annotation encoding="application/x-tex">q</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.19444em"></span><span class="mord mathdefault" style="margin-right:0.03588em">q</span></span></span></span></span> is the frequency of the second allele. The formula describes the frequency of all possible genotypes where</p><table><thead><tr><th align="center">HWE variable</th><th align="center">Genotype</th><th align="center">State</th></tr></thead><tbody><tr><td align="center"><span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>p</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">p^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">p</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span></td><td align="center">&quot;pp&quot;</td><td align="center">homozygous</td></tr><tr><td align="center"><span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>q</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">q^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.03588em">q</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span></td><td align="center">&quot;qq&quot;</td><td align="center">homozygous</td></tr><tr><td align="center"><span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2</mn><mi>p</mi><mi>q</mi></mrow><annotation encoding="application/x-tex">2pq</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8388800000000001em;vertical-align:-0.19444em"></span><span class="mord">2</span><span class="mord mathdefault">p</span><span class="mord mathdefault" style="margin-right:0.03588em">q</span></span></span></span></span></td><td align="center">&quot;pq&quot;</td><td align="center">heterozygous</td></tr></tbody></table><p>Testing for deviation from HWE is usually done with a Chi-Squared (<span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">\chi^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span>) test, where one compares the observed genotype frequencies to the expected genotype frequencies given the observed allele frequencies at a locus. Specifically, the equation is</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∑</mo><mfrac><mrow><mo stretchy="false">(</mo><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi><mo>−</mo><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi><msup><mo stretchy="false">)</mo><mn>2</mn></msup></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi></mrow></mfrac></mrow><annotation encoding="application/x-tex">\sum{\frac{(observed - expected)^2}{expected}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:2.371548em;vertical-align:-0.8804400000000001em"></span><span class="mop op-symbol large-op" style="position:relative;top:-0.000004999999999977245em">∑</span><span class="mspace" style="margin-right:0.16666666666666666em"></span><span class="mord"><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.491108em"><span style="top:-2.314em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord mathdefault">e</span><span class="mord mathdefault">x</span><span class="mord mathdefault">p</span><span class="mord mathdefault">e</span><span class="mord mathdefault">c</span><span class="mord mathdefault">t</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span></span></span><span style="top:-3.23em"><span class="pstrut" style="height:3em"></span><span class="frac-line" style="border-bottom-width:0.04em"></span></span><span style="top:-3.677em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mopen">(</span><span class="mord mathdefault">o</span><span class="mord mathdefault">b</span><span class="mord mathdefault">s</span><span class="mord mathdefault">e</span><span class="mord mathdefault" style="margin-right:0.02778em">r</span><span class="mord mathdefault" style="margin-right:0.03588em">v</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span><span class="mspace" style="margin-right:0.2222222222222222em"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222222222222222em"></span><span class="mord mathdefault">e</span><span class="mord mathdefault">x</span><span class="mord mathdefault">p</span><span class="mord mathdefault">e</span><span class="mord mathdefault">c</span><span class="mord mathdefault">t</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span><span class="mclose"><span class="mclose">)</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.8804400000000001em"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></span></div><p>where <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi></mrow><annotation encoding="application/x-tex">observed</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em"></span><span class="mord mathdefault">o</span><span class="mord mathdefault">b</span><span class="mord mathdefault">s</span><span class="mord mathdefault">e</span><span class="mord mathdefault" style="margin-right:0.02778em">r</span><span class="mord mathdefault" style="margin-right:0.03588em">v</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span></span></span></span></span> is the observed genotype frequency and <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi></mrow><annotation encoding="application/x-tex">expected</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em"></span><span class="mord mathdefault">e</span><span class="mord mathdefault">x</span><span class="mord mathdefault">p</span><span class="mord mathdefault">e</span><span class="mord mathdefault">c</span><span class="mord mathdefault">t</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span></span></span></span></span> is the expected genotype frequency for a locus. To generate our test statistic, we calculate the degrees of freedom: </p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>d</mi><mi>e</mi><mi>g</mi><mi>r</mi><mi>e</mi><mi>e</mi><mi>s</mi><mtext> </mtext><mi>o</mi><mi>f</mi><mtext> </mtext><mi>f</mi><mi>r</mi><mi>e</mi><mi>e</mi><mi>d</mi><mi>o</mi><mi>m</mi><mo>=</mo><msub><mi>n</mi><mrow><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi><mtext> </mtext><mi>g</mi><mi>e</mi><mi>n</mi><mi>o</mi><mi>t</mi><mi>y</mi><mi>p</mi><mi>e</mi><mi>s</mi></mrow></msub><mo>−</mo><msub><mi>n</mi><mrow><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi><mtext> </mtext><mi>a</mi><mi>l</mi><mi>l</mi><mi>e</mi><mi>l</mi><mi>e</mi><mi>s</mi></mrow></msub></mrow><annotation encoding="application/x-tex">degrees\ of\ freedom = n_{expected\ genotypes} - n_{observed\ alleles}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em"></span><span class="mord mathdefault">d</span><span class="mord mathdefault">e</span><span class="mord mathdefault" style="margin-right:0.03588em">g</span><span class="mord mathdefault" style="margin-right:0.02778em">r</span><span class="mord mathdefault">e</span><span class="mord mathdefault">e</span><span class="mord mathdefault">s</span><span class="mspace"> </span><span class="mord mathdefault">o</span><span class="mord mathdefault" style="margin-right:0.10764em">f</span><span class="mspace"> </span><span class="mord mathdefault" style="margin-right:0.10764em">f</span><span class="mord mathdefault" style="margin-right:0.02778em">r</span><span class="mord mathdefault">e</span><span class="mord mathdefault">e</span><span class="mord mathdefault">d</span><span class="mord mathdefault">o</span><span class="mord mathdefault">m</span><span class="mspace" style="margin-right:0.2777777777777778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em"></span></span><span class="base"><span class="strut" style="height:0.8694379999999999em;vertical-align:-0.286108em"></span><span class="mord"><span class="mord mathdefault">n</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3361079999999999em"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">x</span><span class="mord mathdefault mtight">p</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">c</span><span class="mord mathdefault mtight">t</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">d</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathdefault mtight" style="margin-right:0.03588em">g</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">n</span><span class="mord mathdefault mtight">o</span><span class="mord mathdefault mtight">t</span><span class="mord mathdefault mtight" style="margin-right:0.03588em">y</span><span class="mord mathdefault mtight">p</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">s</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.286108em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2222222222222222em"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222222222222222em"></span></span><span class="base"><span class="strut" style="height:0.58056em;vertical-align:-0.15em"></span><span class="mord"><span class="mord mathdefault">n</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.33610799999999996em"><span style="top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight">o</span><span class="mord mathdefault mtight">b</span><span class="mord mathdefault mtight">s</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight" style="margin-right:0.02778em">r</span><span class="mord mathdefault mtight" style="margin-right:0.03588em">v</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">d</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathdefault mtight">a</span><span class="mord mathdefault mtight" style="margin-right:0.01968em">l</span><span class="mord mathdefault mtight" style="margin-right:0.01968em">l</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight" style="margin-right:0.01968em">l</span><span class="mord mathdefault mtight">e</span><span class="mord mathdefault mtight">s</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span></span></span></span></span></div><p>and use this as the parameter for our <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">\chi^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span> distribution, followed by a cumulative density function using this <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">\chi^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span> distribution and our <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>χ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">\chi^2</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.008548em;vertical-align:-0.19444em"></span><span class="mord"><span class="mord mathdefault">χ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em"><span style="top:-3.063em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span></span> value calculated above.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="chi-squared-test"></a>Chi-Squared Test<a aria-hidden="true" tabindex="-1" class="hash-link" href="#chi-squared-test" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-julia codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">hwe_test(data::PopData, by::String = &quot;locus&quot;, correction::String = &quot;none&quot;)</span></div></div></div></div></div><p>Calculate chi-squared test of HWE for each locus and returns observed and expected heterozygosity with chi-squared, degrees of freedom and p-values for each locus. Use <code>by = &quot;population&quot;</code> to perform this separately for each population (default: <code>&quot;locus&quot;</code>). Use <code>correction =</code> to specify a P-value correction method for multiple testing (recommended). For convenience, the correction method is appended to the name of the column, so you will always know how those P-values were adjusted.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="arguments"></a>arguments<a aria-hidden="true" tabindex="-1" class="hash-link" href="#arguments" title="Direct link to heading">#</a></h3><ul><li><code>data</code> : the input <code>PopData</code></li><li><code>by_pop =</code> : <code>false</code> (default) or <code>true</code> for hwe-by-population</li><li><code>correction =</code>  : a string specifying a P-value adjustment type (default: &quot;none&quot;)</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="correction-methods"></a><code>correction</code> methods<a aria-hidden="true" tabindex="-1" class="hash-link" href="#correction-methods" title="Direct link to heading">#</a></h3><ul><li><code>&quot;bonferroni&quot;</code> : Bonferroni adjustment</li><li><code>&quot;holm&quot;</code> : Holm adjustment</li><li><code>&quot;hochberg&quot;</code> : Hochberg adjustment</li><li><code>&quot;bh&quot;</code> : Benjamini-Hochberg adjustment</li><li><code>&quot;by&quot;</code> : Benjamini-Yekutieli adjustment</li><li><code>&quot;bl&quot;</code>  : Benjamini-Liu adjustment</li><li><code>&quot;hommel&quot;</code> : Hommel adjustment</li><li><code>&quot;sidak&quot;</code> : Šidák adjustment</li><li><code>&quot;forwardstop&quot;</code> or <code>&quot;fs&quot;</code> : Forward-Stop adjustment</li><li><code>&quot;bc&quot;</code> : Barber-Candès adjustment</li></ul><p>🤔 For more information on multiple testing adjustments, see <a href="https://juliangehring.github.io/MultipleTesting.jl/stable/" target="_blank" rel="noopener noreferrer">MultipleTesting.jl</a></p><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="example"></a>example<a aria-hidden="true" tabindex="-1" class="hash-link" href="#example" title="Direct link to heading">#</a></h3><div><ul role="tablist" aria-orientation="horizontal" class="tabs tabs--block"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_3zTA tabs__item--active">HWE Chi-Sq</li><li role="tab" tabindex="0" aria-selected="false" class="tabs__item tabItem_3zTA">HWE with P adjustment</li><li role="tab" tabindex="0" aria-selected="false" class="tabs__item tabItem_3zTA">HWE with by population</li></ul><div role="tabpanel" class="margin-vert--md"><div><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-undefined codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">julia&gt; hwe_test(gulfsharks())</span></div><div class="token-line" style="color:#393A34"><span class="token plain">2213×4 DataFrame</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ Row  │ locus        │ chisq     │ df    │ P        │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│      │ Categorical… │ Float64   │ Int64 │ Float64? │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">├──────┼──────────────┼───────────┼───────┼──────────┤</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 1    │ contig_35208 │ 0.690979  │ 3     │ 0.875324 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2    │ contig_23109 │ 1.71251   │ 1     │ 0.19066  │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 3    │ contig_4493  │ 2.0785    │ 1     │ 0.149387 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 4    │ contig_10742 │ 0.249703  │ 1     │ 0.617284 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 5    │ contig_14898 │ 2.10859   │ 1     │ 0.146474 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 6    │ contig_8483  │ 0.019229  │ 1     │ 0.889712 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">⋮</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2207 │ contig_18959 │ 0.968083  │ 1     │ 0.325159 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2208 │ contig_43517 │ 0.655517  │ 1     │ 0.418147 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2209 │ contig_27356 │ 1.71412   │ 1     │ 0.190451 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2210 │ contig_475   │ 0.0754715 │ 1     │ 0.783531 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2211 │ contig_19384 │ 2.10124   │ 1     │ 0.147179 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2212 │ contig_22368 │ 0.473923  │ 1     │ 0.491188 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2213 │ contig_2784  │ 0.0452162 │ 1     │ 0.831607 │</span></div></div></div></div></div></div></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="interpreting-the-results"></a>Interpreting the results<a aria-hidden="true" tabindex="-1" class="hash-link" href="#interpreting-the-results" title="Direct link to heading">#</a></h2><p>Since the results are in table form, you can easily process the table using <code>DataFramesMeta.jl</code> or <code>Query.jl</code> to find loci above or below the alpha threshold you want. As an example, let&#x27;s perform an HWE-test on the <code>nancycats</code> data without any P-value adjustments:</p><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-julia codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">julia&gt; ncats_hwe = hwe_test(nancycats() , by = &quot;population&quot;) ;</span></div></div></div></div></div><p>Now, we can use <code>DataFramesMeta.jl</code> to easily filter this table and leave only what we&#x27;re interested in:</p><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-julia codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">using DataFramesMeta</span></div><div class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#393A34"><span class="token plain">julia&gt; @where(ncats_hwe, :P .!== missing, :P .&lt;= 0.05)</span></div></div></div></div></div><p>With this command, we invoke the <code>@where</code> filtering macro, then specify our <code>ncats_hwe</code> table, the <code>:P</code> column of P-values, and then specify two filtering parameters: </p><ol><li>the P-values are not <code>missing</code></li><li>the P-values are less than or equal to 0.05. </li></ol><p>Doing this results in a table that now only includes non-missing P-values of 0.05 or lower:</p><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-undefined codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">46×5 DataFrame</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ Row │ locus │ population │ chisq   │ df    │ P           │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│     │ Cat…  │ Cat…       │ Float64 │ Int64 │ Float64?    │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">├─────┼───────┼────────────┼─────────┼───────┼─────────────┤</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 1   │ fca8  │ 2          │ 29.8624 │ 15    │ 0.0124275   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 2   │ fca8  │ 3          │ 39.72   │ 21    │ 0.00804172  │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 3   │ fca8  │ 4          │ 64.3039 │ 45    │ 0.0308567   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 4   │ fca8  │ 6          │ 34.0525 │ 21    │ 0.0357734   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 5   │ fca8  │ 7          │ 37.6531 │ 15    │ 0.00101511  │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 6   │ fca23 │ 1          │ 26.0494 │ 10    │ 0.00367434  │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">⋮</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 40  │ fca96 │ 14         │ 58.0854 │ 36    │ 0.0112879   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 41  │ fca96 │ 15         │ 26.51   │ 15    │ 0.0329918   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 42  │ fca96 │ 2          │ 92.0716 │ 15    │ 4.06675e-13 │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 43  │ fca37 │ 1          │ 10.1562 │ 3     │ 0.0172836   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 44  │ fca37 │ 3          │ 13.4815 │ 6     │ 0.0359962   │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 45  │ fca37 │ 5          │ 30.0206 │ 6     │ 3.89561e-5  │</span></div><div class="token-line" style="color:#393A34"><span class="token plain">│ 46  │ fca37 │ 8          │ 33.5111 │ 21    │ 0.0408481   │</span></div></div></div></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="visualizing-the-results"></a>Visualizing the results<a aria-hidden="true" tabindex="-1" class="hash-link" href="#visualizing-the-results" title="Direct link to heading">#</a></h3><p>While not strictly necessary, it might sometimes make sense to generate of heatmap of the results for easier visualization. This is feasible for the <code>nancycats</code> data, but when loci are in the hundreds or thousands, this method quickly becomes counterproductive. In any case, here is a simple example of the HWE results for <code>nancycats</code> plotted as a heatmap using <a href="https://github.com/queryverse/VegaLite.jl">VegaLite.jl</a>:</p><div class="mdxCodeBlock_iHAB"><div class="codeBlockContent_32p_"><button type="button" aria-label="Copy code to clipboard" class="copyButton_1BYj">Copy</button><div tabindex="0" class="prism-code language-julia codeBlock_19pQ"><div class="codeBlockLines_2n9r" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">using VegaLite</span></div><div class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#393A34"><span class="token plain">julia&gt; ncats_hwe = hwe_test(nancycats() , by = &quot;population&quot;, correction = &quot;bonferroni&quot;);</span></div><div class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#393A34"><span class="token plain">julia&gt; ncats |&gt; @vlplot(:rect, :locus, :population, color=:P_bonferroni)</span></div></div></div></div></div><p><img src="/PopGen.jl/img/hwe_test.png" alt="hwe_test"></p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/pdimens/popgen.jl/edit/documentation/docs/analyses/hardyweinberg.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col text--right"><em><small>Last updated on <time datetime="2020-06-03T16:58:02.000Z" class="docLastUpdatedAt_1sqk">6/3/2020</time></small></em></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/PopGen.jl/docs/tutorials/viewsort"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Viewing and sorting</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/PopGen.jl/docs/analyses/relatedness"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Relatedness (Kinship) »</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_TbNY"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#chi-squared-test" class="table-of-contents__link">Chi-Squared Test</a><ul><li><a href="#arguments" class="table-of-contents__link">arguments</a></li><li><a href="#correction-methods" class="table-of-contents__link"><code>correction</code> methods</a></li><li><a href="#example" class="table-of-contents__link">example</a></li></ul></li><li><a href="#interpreting-the-results" class="table-of-contents__link">Interpreting the results</a><ul><li><a href="#visualizing-the-results" class="table-of-contents__link">Visualizing the results</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="text--center"><div>Copyright © 2020 Pavel Dimens & Jason Selwyn. Built with Docusaurus. Rawr!</div></div></div></footer></div>
<script src="/PopGen.jl/styles.da9164d3.js"></script>
<script src="/PopGen.jl/runtime~main.512265a5.js"></script>
<script src="/PopGen.jl/main.5219d249.js"></script>
<script src="/PopGen.jl/1.4e959812.js"></script>
<script src="/PopGen.jl/2.23b37f7c.js"></script>
<script src="/PopGen.jl/3.d4ef4d3f.js"></script>
<script src="/PopGen.jl/1be78505.c8cbe7a4.js"></script>
<script src="/PopGen.jl/67.d7d524a0.js"></script>
<script src="/PopGen.jl/72e583ac.c15c2edd.js"></script>
<script src="/PopGen.jl/731bcebf.924639df.js"></script>
</body>
</html>